### PR TITLE
#261 [Grid] Pagination and basic Filter. 

### PR DESCRIPTION
**Pull request by martineiber**
## Changes in this pull request
Part of #261

### CODE REVIEW

File: src/Asset/Service/AssetService.php [hunk 1]
```diff
@@ -110,6 +110,25 @@ public function getAsset(int $id): Asset|Archive|Audio|Document|Folder|Image|Tex
...
 
+    /**
+     * @throws SearchException|NotFoundException
+     */
+    public function getAssetFolder(int $id): Folder
+    {
+        $asset = $this->assetSearchService->getAssetById($id);
+
+        if (!$asset instanceof Folder) {
+            throw new NotFoundException('folder', $id);
```
Comment (lukmzig) [inline src/Asset/Service/AssetService.php]: We could use here enum value :)
https://github.com/pimcore/studio-backend-bundle/blob/1.x/src/Util/Constants/ElementTypes.php#L34

File: doc/03_Grid.md [hunk 2]
```diff
@@ -11,3 +11,12 @@ A column configuration represents how the column should behave, for example if i
...
 A column data is the actual data for a column. It has a reference to the column and the actual data.
+
+
+## Filter
+A filter is a way to filter the data in the grid. One Property of getting the grid data is the `filter` property.
+Here you can define `page`, `pageSize` and `includeDescendants`.
+
+`page` is the page number of the data you want to get. 
+`pageSize` is the number of items you want to get. `
+includeDescendants` is a boolean value to include the descendants of the current item.
```
Comment (lukmzig) [inline doc/03_Grid.md]:
```suggestion
`includeDescendants` is a boolean value to include the descendants of the current item.
```

File: src/DataIndex/Grid/GridSearch.php [hunk 3]
```diff
@@ -16,34 +16,45 @@
...
     public function __construct(
-        private SecurityServiceInterface $securityService,
+        private FilterServiceProviderInterface $filterServiceProvider,
         private AssetSearchServiceInterface $assetSearchService,
+        private AssetServiceInterface $assetService
     ) {
     }
 
+    /**
+     * @throws NotFoundException
```
Comment (lukmzig) [inline src/DataIndex/Grid/GridSearch.php]:
```suggestion
     * @throws SearchException|NotFoundException
```

File: src/Grid/Service/GridService.php [hunk 4]
```diff
@@ -77,26 +77,24 @@ public function __construct(
...
-            $type = $item->getElementType()->value;
-            $asset = $this->getElement($this->serviceResolver, $type, $item->getId());
-
+        foreach ($items as $item) {
+            $asset = $this->getElement($this->serviceResolver, 'asset', $item->getId());
             $data[] = $this->getGridDataForElement(
                 $this->getConfigurationFromArray($gridParameter->getColumns()),
                 $asset,
-                $type
+                'asset'
```
Comment (lukmzig) [inline src/Grid/Service/GridService.php]: We can also use element types enum here :)

**Review by lukmzig — CHANGES_REQUESTED** [review]
Some small changes, the rest :seal:

File: src/Asset/Service/AssetService.php (hunk 1 above)
Comment (martineiber) [reply src/Asset/Service/AssetService.php]: Thanks for the hint.. Changed it :)

**Review by lukmzig — APPROVED** [review]

**Pull request merged**